core-bundle/contao/library/Contao/DcaLoader.php

```diff
@@ -72,6 +72,11 @@ public function load()
 				throw static::$arrLoaded['dcaFiles'][$this->strTable];
 			}
 
+			if (!isset($GLOBALS['TL_DCA'][$this->strTable]))
+			{
+				trigger_deprecation('contao/core-bundle', '5.0', 'Loading a non-existent DCA has has been deprecated and will throw an exception in Contao 6.0.');
+			}
```

Comment:
This would have helped debugging the issue IMO because it woud’ve detected DCA recursions.